node/node.go

```diff
@@ -19,13 +25,22 @@ var log = logging.Logger("node")
 type Node struct {
 	Type   Type
 	Config *Config
-	Host   host.Host
 
 	// the Node keeps a reference to the DI App that controls the lifecycles of services registered on the Node.
 	app *fx.App
 
-	// the RPC client is only necessary for full nodes
-	RPCClient *rpc.Client `optional:"true"`
+	// CoreClient provides access to Core Node
+	CoreClient core.Client `optional:"true"`
+
+	// p2p components
+	ID          peer.ID
+	Host        host.Host
+	PeerStore   peerstore.Peerstore
+	ConnManager connmgr.ConnManager
```

Comment:
This info is available on `Host` -- let's try not to store duplicate info when we can reach it via Host -- instead we can just make methods on host or something.

Reply:
Should be resolved together with https://github.com/celestiaorg/celestia-node/pull/57#discussion_r708380335